users/src/main/scala/users/domain/User.scala

```diff
@@ -0,0 +1,129 @@
+package users.domain
+
+import java.time.OffsetDateTime
+
+import cats.kernel.Eq
+import cats.implicits._
+import com.softwaremill.quicklens._
+
+final case class User(
+    id: User.Id,
+    userName: UserName,
+    emailAddress: EmailAddress,
+    password: Option[Password],
+    metadata: User.Metadata
+) {
+  def status: User.Status =
+    User.status(this)
+
+  def isActive: Boolean = status === User.Status.Active
+  def isBlocked: Boolean = status === User.Status.Blocked
+  def isDeleted: Boolean = status === User.Status.Deleted
+
+  def updateEmailAddress(emailAddress: EmailAddress, at: OffsetDateTime): User =
+    User.updateEmailAddress(this, emailAddress, at)
+
+  def updatePassword(password: Password, at: OffsetDateTime): User =
+    User.updatePassword(this, password, at)
+
+  def resetPassword(at: OffsetDateTime): User =
+    User.resetPassword(this, at)
+
+  def block(at: OffsetDateTime): User =
+    User.block(this, at)
+
+  def unblock(at: OffsetDateTime): User =
+    User.unblock(this, at)
+
+  def delete(at: OffsetDateTime): User =
+    User.delete(this, at)
+}
+
+object User {
+  def apply(
+      id: User.Id,
+      userName: UserName,
+      emailAddress: EmailAddress,
+      password: Option[Password],
+      at: OffsetDateTime
+  ): User = User(id, userName, emailAddress, password, Metadata(1, at, at, None, None))
+
+  final case class Id(value: String) extends AnyVal
```

Comment:
`String` -> `UUID`

Might be interesting to see how later in the API the mapping from parameter to type is done, and how non-UUID input is handled.

Reply:
Interesting, yes, but at the same time a detail that I don't want them to spend another 30 minutes on.